app/client/cypress/integration/home/glossary.toggle.spec.js

```diff
@@ -0,0 +1,19 @@
+describe('Homepage', () => {
+  beforeEach(() => {
+    cy.visit('/');
+  });
+  it(`Glossary button toggles slide content`, () => {
+    cy.get('button[title="Glossary"]').click();
+  });
+  it(`Glossary close button closes slide content`, () => {
+    cy.get('body').then(($body) => {
+      if ($body.find('button[title="Close glossary"]').length > 0) {
+        cy.get('button[title="Close glossary"]').then(($header) => {
+          if ($header.is(':visible')) {
+            cy.findByText('Close glossary').click();
+          }
+        });
+      }
+    });
+  });
+});
```

Comment:
In the first test, you're not currently asserting that the glossary button toggles in the content – only that the element exists (and can take a click event). Also use `dom-testing-library` selectors over `cy.get()`. You'll need to use `findAllByText('Glossary')` since the text "Glossary" is in the DOM twice – once in the button, and once not visible in the panel that will slide over, so you'll then need to filter for only the visible element. Then you'll finally want to assert that the panel has shown, which I'm doing with `findByPlaceholderText()`.

In the second test, you're not actually testing the close functionality because you're not opening up the panel first. You can get the close button by its text, since it's a small unicode "×" character (different from the alpha "x").

```suggestion
describe('Homepage', () => {
  beforeEach(() => {
    cy.visit('/');
  });

  it(`Glossary button toggles slide content`, () => {
    cy.findAllByText('Glossary')
      .filter(':visible')
      .click();
    cy.findByPlaceholderText('Search for a term...').should('be.visible');
  });

  it(`Glossary close button closes slide content`, () => {
    cy.findAllByText('Glossary')
      .filter(':visible')
      .click();
    cy.findByText('×').click();
    cy.findByPlaceholderText('Search for a term...').should('not.be.visible');
  });
});

```

Reply:
@courtneymyers When I try suggestion it fails with: "CypressError: Timed out retrying: expected '<input.sc-fzXfLV.kPxpBI.js-glossary-search.form-control>' to be 'visible'
This element '<input.sc-fzXfLV.kPxpBI.js-glossary-search.form-control>' is not visible because its ancestor has 'position: fixed' CSS property and it is overflowed by other elements. How about scrolling to the element with cy.scrollIntoView()?" 
Curious if you had successful test with that. Trying to use their suggestion with cy.scrollIntoView() but haven't gotten working yet.

Reply:
@courtneymyers I found solution with scrollIntoView that passes, you can review it.

Reply:
Thanks, @julmcbride. I'm going to take a closer look at this... I think it's a timing issue, as I got the error message the first time I ran the test (after reading your first comment), but not when I re-ran them immediately after. And I never got that error when I first left that code suggestion.

While adding in `cy.scrollIntoView()` makes the test pass, we don't want to have Cypress do any scrolling the user wouldn't normally do – I think that's a situation where the error message Cypress provides goes against the testing philosophy of us writing tests that replicate a real user using the app.

I'm going to look into the timing issue... I think we'll want to wait for the the glossary constructor to do it's initialization before we try clicking the button.